
syntax = "proto3";

package helloworld;

service Greeter {
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc SayHelloAgain (HelloRequest) returns (HelloList) {}
}

message HelloRequest {
  reserved 2;
  uint32 id = 1;
  string name = 3;
}

message detail {
  uint32 status = 1;
  string name = 4;
}
message HelloReply {
  uint32 id = 1;
  string name = 4;
  map<int32, detail> emplyee_code = 5;
}

message HelloList{
repeated HelloReply data = 1;
}